rest/client/methods.go

```diff
@@ -258,13 +258,9 @@ func (c *communicatorImpl) GetNextTask(ctx context.Context, details *apimodels.G
 	info.path = "agent/next_task"
 	resp, err := c.retryRequest(ctx, info, details)
 	if err != nil {
-		err = errors.Wrap(err, "failed to get task")
-		return nil, err
+		return nil, errors.Wrap(err, "failed to get task")
 	}
 	defer resp.Body.Close()
-	if resp.StatusCode == http.StatusConflict {
-		return nil, errors.New("conflict - wrong secret")
-	}
```

Comment:
doesn't this mean that we will retry on conflicts?

Reply:
No, this was dead code. `retryRequest` returns an error on anything that isn't 200, so once we get here it can't be a 409.